src/main/java/jenkins/plugins/git/GitPublisherStep.java

```diff
@@ -0,0 +1,367 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2016.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.plugins.git;
+
+import com.google.inject.Inject;
+import hudson.*;
+import hudson.model.*;
+import hudson.plugins.git.*;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import hudson.plugins.git.extensions.impl.LocalBranch;
+import hudson.scm.SCM;
+import hudson.util.ListBoxModel;
+import java.util.*;
+import org.eclipse.jgit.transport.RemoteConfig;
+import org.eclipse.jgit.transport.URIish;
+import org.jenkinsci.plugins.gitclient.GitClient;
+import org.jenkinsci.plugins.gitclient.PushCommand;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+/**
+ * Runs Git Publishing steps using {@link GitSCM}.
+ */
+public final class GitPublisherStep extends AbstractStepImpl {
+
+  private final String url;
+  private String branch = "master";
+  private String credentialsId;
+
+  private boolean forcePush;
+
+  private List<GitPublisher.TagToPush> tagsToPush;
+  // Pushes HEAD to these locations
+  private List<GitPublisher.BranchToPush> branchesToPush;
+  // notes support
+  private List<GitPublisher.NoteToPush> notesToPush;
+
+  @DataBoundConstructor
+  public GitPublisherStep(String url) {
+    this.url = url;
+  }
+
+  public String getUrl() {
+    return url;
+  }
+
+  public String getBranch() {
+    return branch;
+  }
+
+  public String getCredentialsId() {
+    return credentialsId;
+  }
+
+  @DataBoundSetter
+  public void setBranch(String branch) {
```

Comment:
I would recommend to make these setters `@Restricted(NoExternalUse.class)`. Otherwise it worth to mention in Javadoc that invocation of these setters does not persist data to the disk.